### PR TITLE
Fix outputting correct information for nonformatted source

### LINKED ISSUE #39: Wrong output when the file isn't gofmt'd
With the following file:

```
$ cat tags.go
package a
type x struct {
    Foo int
    bar int
}
```

The tool gives the wrong results.

Should be start=2:

```
$ gomodifytags -format json -file tags.go -transform snakecase -line 3,4 -add-tags json -add-options json=
{
  "start": 3,
  "end": 4,
  "lines": [
    "type x struct {",
    "\tFoo int `json:\"foo\"`"
  ]
}
```

Should be start=2, and one line is missing from the output:

```
$ gomodifytags -format json -file tags.go -transform snakecase -line 3,3 -add-tags json -add-options json=
{
  "start": 3,
  "end": 3,
  "lines": [
    "type x struct {"
  ]
}
```

start is correct, but the last line (`}`) missing:
```
$ gomodifytags -format json -file tags.go -transform snakecase -offset 23 -add-tags json -add-options json=
{
  "start": 2,
  "end": 5,
  "lines": [
    "",
    "type x struct {",
    "\tFoo int `json:\"foo\"`",
    "\tbar int `json:\"bar\"`"
  ]
}
```

It seems that the tool operates on the file *as if* they're gofmt'd.

### PR DESCRIPTION
**Pull request by fatih**
This makes sure to return the correct lines for the `--json` format if
the given input is not formatted (gofmt'd).

fixes #39

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -480,20 +481,44 @@ func (c *config) format(file ast.Node, rwErrs error) (string, error) {
...
+
+			// check for any line directive and store it for next iteration to
+			// re-construct the original file
+			if !strings.HasPrefix(txt, "//line") {
+				lines = append(lines, txt)
+				continue
+			}
+
+			line := strings.Split(txt, ":")
+			lineNr, err := strconv.Atoi(line[1])
```
Comment (bouk) [inline main.go]: This will panic if it looks like this:

```
//line blabla.txt
```

(Don’t know if that can happen)

Comment (fatih) [reply main.go]: Yeah! I fixed it locally but forget to push it. Thanks for reminding :)

**Comment by josa42** [comment]
@fatih I noticed another edge case: If first line in the struct is empty the indentation (tab) gets replaced by a single space.

Source:

```
type x struct {

	Foo int
}
```

Result:
```
type x struct {

 Foo int `json:"foo"`
}
```

## Test Case
See: [josa42/gomodifytags `a609848`](https://github.com/josa42/gomodifytags/commit/a6098484b5d9c5d628b9ae04ad0f3e4800d513d6)

<details><summary>Test</summary>

```
{
	file: "json_not_formatted_7",
	cfg: &config{
		add:  []string{"json"},
		line: "2,6",
	},
}
```

`json_not_formatted_7.input`

```go
package main

type x struct {

	Foo int
}
```

`json_not_formatted_7.golden`
```json
{
  "start": 2,
  "end": 6,
  "lines": [
    "",
    "type x struct {",
    "",
    "\tFoo int `json:\"foo\"`",
    "}"
  ]
}
```

</details>
<details><summary>Result</summary>

```
got:
====

{
  "start": 2,
  "end": 6,
  "lines": [
    "",
    "type x struct {",
    "",
    " Foo int `json:\"foo\"`",
    "}"
  ]
}
want:
=====

{
  "start": 2,
  "end": 6,
  "lines": [
    "",
    "type x struct {",
    "",
    "\tFoo int `json:\"foo\"`",
    "}"
  ]
}

from:
=====

package main

type x struct {

    Foo int
}

```

</details>

**Comment by fatih** [comment]
@josa42 Thanks for the feedback. You're right. While implementing this fix, I've also seen this behavior. This is due the parser and printer. I think the parser has a bug when the `SourcePos` is enabled and strips `\t` characters. The output is syntactically correct though and if you gofmt it it'll be fixed automatically.

**Pull request merged**